### DISCUSSION TITLE
How to add strata to cox Regularized Cox Regression?

### DISCUSSION THREAD

**Discussion by hippo-yf** [mlr-org/mlr3proba#425] [Q&A] [answered]
Hello,
    How to add strata (different hazard baseline)  to survival regression in  Task **TskSurv** with learner **surv.cv_glmnet** ? In original glmnet package, **stratifySurv** is used to add strata. But , in mlr3proba, I do not find the setting.  Thanks.

**Comment by bblodfon** [comment]
Hi @hippo-yf,

Thanks for mentioning this, I think it can be done, so transfering to `mlr3extralearners` (but it's going to be `glmnet`-specific). 
See issue here => https://github.com/mlr-org/mlr3extralearners/issues/396.

**Comment by bblodfon** [comment, marked as answer]
Hi Yance (@hippo-yf), this feature is implemented in https://github.com/mlr-org/mlr3extralearners/pull/607 and will soon be merged in the development github version of `mlr3extralearners`. Simple example:


``` r
library(mlr3proba)
#> Loading required package: mlr3
library(mlr3extralearners)

task = tsk("gbcs")
train_rows = 1:600
test_rows = 601:task$nrow

learner = lrn("surv.glmnet", lambda = 0.03, strata = "hormone")
learner$train(task, train_rows)
learner$predict(task, test_rows)
#> 
#> ── <PredictionSurv> for 86 observations: ───────────────────────────────────────
#>  row_ids time status     crank        lp     distr
#>      601 1443  FALSE 1.1663890 1.1663890 <list[1]>
#>      602 1317  FALSE 0.3946265 0.3946265 <list[1]>
#>      603 1344  FALSE 0.4068678 0.4068678 <list[1]>
#>      ---  ---    ---       ---       ---       ---
#>      684  768  FALSE 3.9682253 3.9682253 <list[1]>
#>      685  858  FALSE 0.5609451 0.5609451 <list[1]>
#>      686  770  FALSE 0.8926821 0.8926821 <list[1]>
```

<sup>Created on 2026-08-03 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>
